corehq/warehouse/models/facts.py

```diff
@@ -0,0 +1,108 @@
+from django.db import models, transaction, connections
+
+from corehq.warehouse.const import (
+    APP_STATUS_FACT_SLUG,
+    FORM_FACT_SLUG,
+    USER_DIM_SLUG,
+    DOMAIN_DIM_SLUG,
+    FORM_STAGING_SLUG,
+    SYNCLOG_STAGING_SLUG,
+)
+
+from .dimensions import UserDim, DomainDim
+from corehq.form_processor.models import XFormInstanceSQL
+from corehq.sql_db.routers import db_for_read_write
+from corehq.util.test_utils import unit_testing_only
+from corehq.warehouse.etl import CustomSQLETLMixin
+from corehq.warehouse.models.shared import WarehouseTableMixin
+
+
+class BaseFact(models.Model, WarehouseTableMixin):
+
+    @classmethod
+    @transaction.atomic
```

Comment:
since this stuff happens in the non-default db we'll actually need to do `with transaction.atomic(using=database):`